pareto/utilities/results.py

```diff
@@ -1151,50 +1177,140 @@ def generate_report(
-                                    (*i, var_value)
-                                )
-                        else:
-                            headers[str(variable.name) + "_dict"].append(
-                                (*i, var_value)
-                            )
+                        continue
+
+                    # if a variable contains only one index, then "i" is recognized as a string and not a tuple,
+                    # in that case, "i" is redefined by adding a comma so that it becomes a tuple
+                    if isinstance(i, str):
+                        i = (i,)
+
+                    # replace the discrete qualities by their actual values
+                    if str(variable.name) == "v_DQ":
+                        var_value = model.p_discrete_quality[i[2], i[3]].value
+
+                    if var_value is not None and variable.name not in [
+                        "inlet_salinity",
+                        "v_C_TreatmentCapEx_site",
+                        "v_C_Treatment_site",
+                        "v_C_Treatment_site_ReLU",
+                        "recovery",
+                        "v_C_TreatmentCapEx_site_time",
+                        "totalCapex",
+                        "v_T_Treatment_scaled",
+                        "v_T_Treatment_scaled_ReLU",
+                    ]:
+                        headers[str(variable.name) + "_dict"].append((*i, var_value))
+
+    # Loop through all the expressions in the model
+    for expr in model.component_objects(Expression):
+        # The get_units function does not work properly when called on an
+        # indexed expression, so we have to grab the units from the first
+        # index when we iterate through the expression. When we do that, we
+        # mark the units_fetched flag as True so we only do it once.
+        units_fetched = False
+        # Loop through the indexes of the expression.
+        for i in expr:
+            if not units_fetched:
+                units_true = (
+                    pyunits.get_units(expr[i]) is not None
+                    and pyunits.get_units(expr[i]).to_string() != "dimensionless"
+                )
+
+                # If units are used, determine what the display units should be based off user input
+                if units_true:
+                    from_unit = pyunits.get_units(expr[i])
+                    from_unit_string = from_unit.to_string()
+                    # the display units (to_unit) is defined by output_units from module parameter
+                    if output_units == OutputUnits.unscaled_model_units:
+                        to_unit = model.model_to_unscaled_model_display_units[
+                            from_unit_string
+                        ]
+                    elif output_units == OutputUnits.user_units:
+                        to_unit = model.model_to_user_units[from_unit_string]
+                    else:
+                        print(
```

Comment:
Should this not be a raise instead of just printing a warning?

Reply:
I think it's fine to leave this as a warning.